climpred/bootstrap.py

```diff
@@ -497,16 +590,63 @@ def bootstrap_compute(
     # get comparison function
     comparison = get_comparison_class(comparison, ALL_COMPARISONS)
 
-    # Perfect Model requires `same_inits` setup.
-    isHindcast = True if comparison in HINDCAST_COMPARISONS else False
+    # Perfect Model requires `same_inits` setup
+    isHindcast = True if comparison.name in HINDCAST_COMPARISONS else False
     reference_alignment = alignment if isHindcast else 'same_inits'
 
-    for i in range(iterations):
-        # resample with replacement
-        smp_hind = _resample(hind, resample_dim)
-        # compute init skill
-        init_skill = compute(
-            smp_hind,
+    if hist is None:  # PM path, use verif = control
+        hist = verif
+
+    # slower path for hindcast and resample_dim init
+    if resample_dim == 'init' and isHindcast:
+        warnings.warn(f'resample_dim=`init` will be slower than resample_dim=`member`.')
+        (
+            bootstrapped_init_skill,
+            bootstrapped_uninit_skill,
+            bootstrapped_pers_skill,
+            pers_output,
+        ) = _bootstrap_hindcast_over_init_dim(
+            hind,
+            hist,
+            verif,
+            dim,
+            resample_dim,
+            iterations,
+            metric,
+            comparison,
+            alignment,
+            compute,
+            reference_compute,
+            resample_uninit,
+            reference_alignment,
+            **metric_kwargs,
+        )
+    else:  # faster resampling skill: first _resample_iterations_idx, then compute skill
+        bootstrapped_hind = _resample_iterations_idx(hind, iterations, resample_dim)
+        if not isHindcast:
+            # create more members than needed in PM to make the uninitialized
+            # distribution more robust
+            members_to_sample_from = 50
+            repeat = members_to_sample_from // hind.member.size + 1
```

Comment:
if you set repeat to 1 there are no errors

Reply:
but then it also obviously doesnt create more than hind.member.size members and I think more might be useful to appropriately sample the control distribution